File: Answer/1680.连接连续二进制数字.cs
/*
 * @lc app=leetcode.cn id=1680 lang=csharp
 *
 * [1680] 连接连续二进制数字
 */

// @lc code=start
public class Solution
{
    public int ConcatenatedBinary(int n)
    {
        const uint MOD = 1000000007;
        uint res = 0;
        for (uint i = 1; i <= n; i++)
        {
            uint t = i;
            while (t > 0)
            {
                t >>= 1;
                res = (res << 1) % MOD;
            }
            res = (res + i) % MOD;
        }
        return (int)res;
    }
}
// @lc code=end

File: Answer/1680.连接连续二进制数字.go
/*
 * @lc app=leetcode.cn id=1680 lang=golang
 *
 * [1680] 连接连续二进制数字
 */

// @lc code=start
func concatenatedBinary(n int) int {
	MOD := 1000000007
	res := 0
	for i := 1; i <= n; i++ {
		t := i
		for t > 0 {
			t = t >> 1
			res = (res << 1) % MOD
		}
		res = (res + i) % MOD
	}
	return res
}

// @lc code=end
